### PR TITLE
Impl Display for metrics

### PR DESCRIPTION
**Pull request by durch**

### CODE REVIEW

**Review by posix4e — APPROVED** [review]

File: src/metrics/gauge.rs [hunk 1]
```diff
@@ -6,12 +6,13 @@
...
 use std::sync::atomic::{AtomicIsize, Ordering};
 use std::sync::Arc;
+use std::fmt::Display;
 
 /// Naive implementation of a `Gauge`.
 #[derive(Debug)]
 pub struct StdGauge {
     /// The gauge value.
-    value: AtomicIsize,
+    pub value: AtomicIsize,
```
Comment (waywardmonkeys) [inline src/metrics/gauge.rs:15]: Does this need to be public?

File: src/metrics/mod.rs [hunk 2]
```diff
@@ -15,6 +15,7 @@ mod meter;
 pub use self::counter::{Counter, CounterSnapshot, StdCounter};
 pub use self::gauge::{Gauge, GaugeSnapshot, StdGauge};
 pub use self::meter::{Meter, MeterSnapshot, StdMeter};
+use std::sync::atomic::Ordering;
```
Comment (waywardmonkeys) [inline src/metrics/mod.rs:18]: Is this needed?

File: src/metrics/meter.rs [hunk 3]
```diff
@@ -20,6 +21,13 @@ pub struct MeterSnapshot {
     pub mean: f64,
 }
 
+impl Display for MeterSnapshot {
+    fn fmt(&self, f: &mut ::std::fmt::Formatter) -> ::std::fmt::Result {
+        write!(f, "{}, {}", self.count, self.mean)
```
Comment (waywardmonkeys) [inline src/metrics/meter.rs:26]: What is this used for? It seems like different things would want to display this in different ways?

Comment (durch) [reply src/metrics/meter.rs:26]: Its so you can do stuff like `println!("{}", metric)` as opposed to `println!("{:?}", metric)`, different things can still display however they want. If you ask why rates is not included, its for sake of simplicity, tough if you'd like I can include it as well...

File: src/metrics/gauge.rs (hunk 1 above)
Comment (durch) [reply src/metrics/gauge.rs:15]: Not at present, I have another PR that will utilize that, a bit of leakage on my part

File: src/metrics/mod.rs (hunk 2 above)
Comment (durch) [reply src/metrics/mod.rs:18]: Same as for the `pub` above

**Comment by posix4e** [comment]
@waywardmonkeys @durch  Anything left here? Am I good to merge?

**Comment by posix4e** [comment]
After further thought, I think for now I don't see a good use case over using debug

**Comment by durch** [comment]
Ok, ill move on to including Futures, in a seperate branch initially

**Pull request closed without merging**